### PR TITLE
Docs: Minor description fix to build procedure on Debian wrt BerkeleyDB

### PR DESCRIPTION
**Pull request by shigeya**
There is some unclear description on this, and I have confused.
Trying to be clear on options.

Verified on Debian 9.3.

### CODE REVIEW

**Comment by dmoney0546** [comment]
People trying to take mybit coins

**Comment by dmoney0546** [comment]
Help

File: doc/build-unix.md [hunk 1]
```diff
@@ -85,22 +85,27 @@ install necessary parts of boost:
...
-    sudo apt-get install libdb4.8-dev libdb4.8++-dev
+        sudo apt-get install libdb-dev libdb++-dev
+  
+  then, pass `--with-incompatible-bdb` to configure without binary compatibility
 
-Ubuntu and Debian have their own libdb-dev and libdb++-dev packages, but these will install
-BerkeleyDB 5.1 or later, which break binary wallet compatibility with the distributed executables which
-are based on BerkeleyDB 4.8. If you do not care about wallet compatibility,
-pass `--with-incompatible-bdb` to configure.
+- Find and install Berkeley DB 4.8 for full binaruy compatibility
```
Comment (practicalswift) [inline doc/build-unix.md]: "Binary" typo :-)

Comment (shigeya) [reply doc/build-unix.md]: :< thanks.

Comment (shigeya) [reply doc/build-unix.md]: Fixed.

**Review by practicalswift — APPROVED** [review]

File: doc/build-unix.md [hunk 2]
```diff
@@ -85,22 +85,27 @@ install necessary parts of boost:
 
         sudo apt-get install libboost-all-dev
 
-BerkeleyDB is required for the wallet.
+BerkeleyDB is required for the wallet. Unfortunately, There is a binary incompatibility between BerkeleyDB 4.8 and latest 5.x series. There are three options:
 
-**For Ubuntu only:** db4.8 packages are available [here](https://launchpad.net/~bitcoin/+archive/bitcoin).
-You can add the repository and install using the following commands:
+- If you need the wallet and don't need the binary wallet compatibility install latest Berkeley DB with following commands:
```
Comment (practicalswift) [inline doc/build-unix.md]: Nit: "Install the latest …"

File: doc/build-unix.md [hunk 3]
```diff
@@ -85,22 +85,27 @@ install necessary parts of boost:
 
         sudo apt-get install libboost-all-dev
 
-BerkeleyDB is required for the wallet.
+BerkeleyDB is required for the wallet. Unfortunately, There is a binary incompatibility between BerkeleyDB 4.8 and latest 5.x series. There are three options:
```
[the remark this answers is not in the document]
Comment (shigeya) [reply doc/build-unix.md]: fixed :< thanks.

File: doc/build-unix.md (hunk 2 above)
Comment (shigeya) [reply doc/build-unix.md]: fixed :< thanks.

**Review by randolf — APPROVED** [review]

File: doc/build-unix.md [hunk 4]
```diff
@@ -85,22 +85,27 @@ install necessary parts of boost:
...
-**For Ubuntu only:** db4.8 packages are available [here](https://launchpad.net/~bitcoin/+archive/bitcoin).
-You can add the repository and install using the following commands:
+- If you need the wallet and don't need the binary wallet compatibility install the latest Berkeley DB with following commands:
 
-    sudo apt-get install software-properties-common
-    sudo add-apt-repository ppa:bitcoin/bitcoin
-    sudo apt-get update
-    sudo apt-get install libdb4.8-dev libdb4.8++-dev
+        sudo apt-get install libdb-dev libdb++-dev
+
```
Comment (AkioNak) [inline doc/build-unix.md]: @shigeya perhaps the whitespace in this line made travis-check to fail.

Comment (shigeya) [reply doc/build-unix.md]: Oh, I see. let me push another one.

**Comment by shigeya** [comment]
Hmm. Looks like intentional trailing space (to adjust indentation of the following text) for formatting cause the issue. I will look into the lint code, but, simply forbid such a trailing space don't work in this context. I tried to rewrite the text prior the `apt-get` commands but can't find a better way to rewrite. Will fix in either way.

**Review by AkioNak — CHANGES_REQUESTED** [review]
I'm sorry, I confused you with my last comment.
I reviewed again with editing my local repository actually.

File: doc/build-unix.md [hunk 5]
```diff
@@ -79,28 +79,33 @@ Options when installing required Boost library files:
...
+  **For Ubuntu only:** db4.8 packages are available [here](https://launchpad.net/~bitcoin/+archive/bitcoin). You can add the repository and install using the following commands:
+
+      sudo apt-get install software-properties-common
+      sudo add-apt-repository ppa:bitcoin/bitcoin
+      sudo apt-get update
+      sudo apt-get install libdb4.8-dev libdb4.8++-dev
+
+- Disabling the wallet
+
```
[the remark this answers is not in the document]
Comment (shigeya) [reply doc/build-unix.md]: Thanks, but that lines does not actually cause the problem. The issue is caused by different lines. Please see the log.

**Comment by shigeya** [comment]
I have modified to eliminate description trailing in `apt-get` code block. I will look into the lint code on spare time in future.

**Comment by fanquake** [comment]
@shigeya Are you planning on following up with the changes needed here?

**Comment by shigeya** [comment]
@fanquake yes. I will. bit busy last two weeks.

**Comment by MarcoFalke** [comment]
Closing for now. You can open a new pull request when you come back to addressing the feedback.

**Pull request closed without merging**